src/alcazar.rs

```diff
@@ -1,42 +1,79 @@
-use crate::http_request::HttpRequest;
-use std::net::{SocketAddr, TcpListener};
+use crate::{
+    http_request::{HttpError, HttpRequest, ParseError},
+    router::Router,
+};
+use std::{
+    io::{Error as IOError, Write},
+    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener},
+};
+use thiserror::Error;
 use tracing::info;
 
-#[derive(Default)]
-pub struct AlcazarBuilder {
-    url: Option<SocketAddr>,
+#[derive(Error, Debug)]
```

Comment:
I think it would be useful for some use cases, to implement `Clone` for errors.

Reply:
I prefere to not implement some code which I am not using, by experience it avoid dead code and bugs.

Reply:
Cloning errors doesn't lead to unxpected behaviour. Cloning mostly used for sending / printing errors in logs or external applications like Sentry

Reply:
Yes but here we are using the `IOError` which is the `std::io::Error` and if we implements the `Clone` trait on `AlcazarError` yet, I have no use case for testing the good behavior of the wrapping which I will have to do for the std Error. This is why I prefere to don't implement some code when I haven't the usecase.

Reply:
Btw I will add the `Clone` trait on `HttpError` and `ParseError` which are ours.